cpu/esp_common/esp-wifi/esp_wifi_netdev.c

```diff
@@ -272,13 +272,16 @@ esp_err_t esp_wifi_internal_reg_rxcb(wifi_interface_t ifx, wifi_rxcb_t fn)
 
     /* now, we have to allocate a new socket and register the function */
     _esp_wifi_socket = esp_socket(AF_PACKET, SOCK_RAW, ETH_P_ALL);
-
+    const char *netcard_name = "sta0";
+    if (ifx == ESP_IF_WIFI_AP) {
+        netcard_name = "ap0";
+    }
     if (_esp_wifi_socket < 0) {
         ESP_WIFI_LOG_ERROR("create socket of (AF_PACKET, SOCK_RAW, ETH_P_ALL) error");
         return ESP_FAIL;
     }
-    if (esp_ioctl(_esp_wifi_socket, SIOCGIFINDEX, "sta0") < 0) {
-        ESP_WIFI_LOG_ERROR("bind socket %d to netcard %s error", _esp_wifi_socket, "sta0");
+    if (esp_ioctl(_esp_wifi_socket, SIOCGIFINDEX, netcard_name) < 0) {
+        ESP_WIFI_LOG_ERROR("bind socket %d to netcard %s error", _esp_wifi_socket, netcard_name);
```

Comment:
This might have been the reason the ESP8266 wasn't receiving any packets. I verified with the esp8266 rtos sdk and `"ap0"` should be used in this case.

Reply:
It works.